<a href="https://colab.research.google.com/github/Halifaxi/Data-Analysis-Project-1/blob/master/Homework_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [17]:
#1c)
import numpy as np 
import matplotlib.pyplot as plt
from numpy import linalg as LA

# W = working, B = broken.

T = np.matrix([[0.7,0.5], [0.3,0.5]])
print("Prob working tomorrow ", np.linalg.matrix_power(T,2)) #tomorrow
print("prob working next day ", np.linalg.matrix_power(T,3))# Next Day
print(" prob working in one week ", np.linalg.matrix_power(T,7)) # one Week
print("prob working in one month " ,np.linalg.matrix_power(T,30)) #one Month


print( )
print()
print(" 1.d) ")

X = np.array([[0.7,0.5], [0.3,0.5]])
[e,v]=LA.eig(X)
print(X)
print(e)
print()
print(v)
v1 = v[:,0]
v1 = v1/sum(v1)
print("Long run distribution: ", v1)




Prob working tomorrow  [[0.64 0.6 ]
 [0.36 0.4 ]]
prob working next day  [[0.628 0.62 ]
 [0.372 0.38 ]]
 prob working in one week  [[0.6250048 0.624992 ]
 [0.3749952 0.375008 ]]
prob working in one month  [[0.625 0.625]
 [0.375 0.375]]


 1.d) 
[[0.7 0.5]
 [0.3 0.5]]
[1.  0.2]

[[ 0.85749293 -0.70710678]
 [ 0.51449576  0.70710678]]
Long run distribution:  [0.625 0.375]


1c) probability of working tomorrow is equal to 64%, probability of working next day is 62.8%, one week is equal to 62.5%, one month is equal to 62.5%. 

1.d) The long run distribution is equal to, W = 62.5% and B = 37.5%. Meaning that the long term probability of the copying machine working at any given day is equal to 62.5%.

In [ ]:
#2c)

T = np.matrix([[.699,.498,0], [.30,.50,0], [.001, .002, 1]])
B = T[2:3,0:2]
A = T[0:2,0:2]
I = np.identity(2)

ll = np.subtract(I,A)
FundaM= np.linalg.inv(ll)
print(T)
print()
print(FundaM)
Final = np.dot(B,FundaM)





[[0.699 0.498 0.   ]
 [0.3   0.5   0.   ]
 [0.001 0.002 1.   ]]

[[454.54545455 452.72727273]
 [272.72727273 273.63636364]]



c) The machine will last on average equal to the sum of column 1 for the fundamental matrix which is equal to around 727.2728 days. 
The amount of that time it will be working is equal to 454.5455 days and the amount of time it will be under repair is equal to 272.7273. This is assuming we are starting in a working computer state. 

In [11]:
#3

T = np.matrix([[0,.6,0,0,0,0,0,0,0],[.4,0,.6,0,0,0,0,0,0], [0,0.4,0,0.6,0,0,0,0,0], [0,0,.4,0,0.6,0,0,0,0], [0,0,0,0.4,0,0.6,0,0,0], [0,0,0,0,0.4,0,0.6,0,0,]
               , [0,0,0,0,0,0.4,0,0,0], [0,0,0,0,0,0,0.4,1,0], [0.6,0,0,0,0,0,0,0,1]])
#print(T)
print()

B = T[7:9,0:7]
#print(B)
print()
A = T[0:7,0:7]
#print(A)
I = np.identity(7)

ll = np.subtract(I,A)
FundaM= np.linalg.inv(ll)
#print(np.matrix(FundaM))
Final = np.dot(B,FundaM)
print(Final)

#b)
X = np.matrix([[0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0], [0,0,0,0,0,0,0,0,0], [0,0,0,0,0,0.6,0,0,0], [0,0,0,0,0,0,0,0,0], [0,0,0.4,0,0,0,0,0,0,]
               , [0,0,0,0,0,0,0,0,0], [0,0,0,0.4,0,0.4,0,1,0], [0,0,0.6,0.6,0,0,0,0,1]])
#print(X)
B = X[7:9,0:7]
#print(B)
print()
A = X[0:7,0:7]
#print(A)
I = np.identity(7)

ll = np.subtract(I,A)
FundaM= np.linalg.inv(ll)
#print(np.matrix(FundaM))
Final = np.dot(B,FundaM)
print(Final)

#d) Bonus:

Z = np.matrix([[0,0,0,0,0,0,0,0,0],[0,0,0.6,0,0,0,0,0,0], [0,0,0,0,0,0,0,0,0], [0,0.4,0.4,0,0,0,0,0,0], [0,0,0,0,0,0,0,0,0], [0,0,0,0,0,0,0,0,0,]
               , [0,0,0,0,0,0,0,0,0], [0,0,0,0.4,0,0,0,1,0], [0,0.6,0,0.6,0,0,0,0,1]])
print()
print(Z)
B = Z[7:9,0:7]
#print(B)
print()
A = Z[0:7,0:7]
#print(A)
I = np.identity(7)
ll = np.subtract(I,A)
FundaM= np.linalg.inv(ll)
#print(np.matrix(FundaM))
Final = np.dot(B,FundaM)
print(Final)









[[0.02030135 0.05075337 0.0964314  0.16494845 0.26772403 0.42188739
  0.65313243]
 [0.97969865 0.94924663 0.9035686  0.83505155 0.73227597 0.57811261
  0.34686757]]

[[0.    0.    0.256 0.4   0.    0.64  0.   ]
 [0.    0.    0.744 0.6   0.    0.36  0.   ]]

[[0.  0.  0.  0.  0.  0.  0.  0.  0. ]
 [0.  0.  0.6 0.  0.  0.  0.  0.  0. ]
 [0.  0.  0.  0.  0.  0.  0.  0.  0. ]
 [0.  0.4 0.4 0.  0.  0.  0.  0.  0. ]
 [0.  0.  0.  0.  0.  0.  0.  0.  0. ]
 [0.  0.  0.  0.  0.  0.  0.  0.  0. ]
 [0.  0.  0.  0.  0.  0.  0.  0.  0. ]
 [0.  0.  0.  0.4 0.  0.  0.  1.  0. ]
 [0.  0.6 0.  0.6 0.  0.  0.  0.  1. ]]

[[0.    0.16  0.256 0.4   0.    0.    0.   ]
 [0.    0.84  0.744 0.6   0.    0.    0.   ]]


3a) The probability he wins 8 dollars is equal to 9.64%. 

b) The probability of winning is 25.6%.

c) Since the probability of success is higher for the bold strategy, Smith's best bet is using the bold strategy.

d bonus) There is no strategy that can beat the bold one, they can only tie it. In the strategy I created, it follows a similar bold strategy, except you start off by betting one dollar instead of all $3. But, there can be no strategy that beats the bold one because if the probability of winning is less than 50%, then the best strategy is the one that gets you out of the game the fastest (least amount of betting turns). No strategy can be created that gets you out of the game in less iterations than the bold strategy given the constraints; so it therefore can only be tied at best. 

In [ ]:
#4 b)

T = np.matrix([[0.2,0,0,0,0,0],[0.7,0.2,0,0,0,0],[0,0.7,0.2,0,0,0],[0,0,0.7,0.2,0,0],[0.1,0.1,0.1,0.1,1,0],[0,0,0,0.7,0,1]])
print(T)
print()

A = T[0:4, 0:4]
#print(A)
B = T[4:6, 0:4]
#print(B)
I = np.identity(4)
#print(I)

ll = np.subtract(I,A)
FundaM= np.linalg.inv(ll)
print(FundaM)
print()
Final = np.dot(B,FundaM)
print(Final)
print()
totalTime= sum(FundaM[:,0])
print(totalTime) # total time spent in medical school




[[0.2 0.  0.  0.  0.  0. ]
 [0.7 0.2 0.  0.  0.  0. ]
 [0.  0.7 0.2 0.  0.  0. ]
 [0.  0.  0.7 0.2 0.  0. ]
 [0.1 0.1 0.1 0.1 1.  0. ]
 [0.  0.  0.  0.7 0.  1. ]]

[[1.25       0.         0.         0.        ]
 [1.09375    1.25       0.         0.        ]
 [0.95703125 1.09375    1.25       0.        ]
 [0.83740234 0.95703125 1.09375    1.25      ]]

[[0.41381836 0.33007812 0.234375   0.125     ]
 [0.58618164 0.66992188 0.765625   0.875     ]]

[[4.13818359]]


4c) The time expected to be in second year is equal 1.09 years. They are expected to be in medical school for about 4.14 years.

4d) The probability of a beginning student flunking out is equal to 0.4138 or around 41%. The probability of graduating is 0.5862 or around 59%.